Source/JavaScriptCore/inspector/InjectedScriptSource.js

```diff
@@ -223,22 +225,37 @@ let InjectedScript = class InjectedScript extends PrototypelessObjectBase
                 try {
                     resolvedArgs[i] = this._resolveCallArgument(callArgs[i]);
                 } catch (e) {
-                    return @String(e);
+                    callback(this._createThrownValue(e, objectGroupName));
+                    return;
                 }
             }
         }
 
         try {
             let func = InjectedScriptHost.evaluate("(" + expression + ")");
-            if (typeof func !== "function")
-                return "Given expression does not evaluate to a function";
-
-            return @createObjectWithoutPrototype(
-                "wasThrown", false,
-                "result", RemoteObject.create(func.@apply(object, resolvedArgs), objectGroupName, returnByValue, generatePreview),
-            );
+            if (typeof func !== "function") {
+                callback(this._createThrownValue("Given expression does not evaluate to a function", objectGroupName));
```

Comment:
ditto for protocol command error behavior change

Reply:
Done.